### ISSUE TITLE
Feature Request: CLI dialog

### ISSUE THREAD

**Issue by mccambridge** [casperjs/casperjs#1037]
It would be great if I could get user input from the command line to proceed conditionally, exit for certain conditions, etc.  For instance, I'd like to create a file that doesn't exist given a certain option.  But first, I want to confirm this with the user.  Something like:

``` javascript
var filename = casper.cli.get('filename');

if (!fs.exists(filename)) {
  // first confirm with user so we're sure there's not a typo
  if (casper.cli.dialog(filename + ' does not exist. Should we create it?') {
    // create file and write there
    ...
  } else {
    // crap out
    casper.warn('Could not write to ' + filename);
    casper.exit();
} else {
  // file exists, so append to it
  ...
}
```

**Comment by kensoh** [comment]
@mccambridge this is an interesting feature. In a project I'm working on now I added a live mode so that I can try steps and JavaScript code to interact with web elements during run-time. Its helps in rapid prototyping.

The PhantomJS execution is ok but Firefox side (through SlimerJS) seems to run into error with reading from standard in even though it is documented. I've raised an issue there for Laurent - https://github.com/laurentj/slimerjs/issues/188

For PhantomJS implementation can use
```js
var sys = require('system');
sys.stdout.write('Your message to user\n');
var input = sys.stdin.readLine();
```
